apps/frontend/src/app/5_pages/ConvertPage/ConvertPage.tsx

```diff
@@ -256,23 +258,29 @@ const ConvertPage: FC = () => {
         return;
       }
 
-      const [sourceTokenDetails, destinationTokenDetails] = await Promise.all([
-        getAssetData(sourceToken, currentChainId),
-        getAssetData(destinationToken, currentChainId),
-      ]);
-
-      const result = await smartRouter.getBestQuote(
-        currentChainId,
-        sourceTokenDetails.address,
-        destinationTokenDetails.address,
-        weiAmount,
-      );
-
-      setRoute(result.route);
-      const quote = removeTrailingZerosFromString(
-        fromWei(result.quote.toString()),
-      );
-      setQuote(quote);
+      try {
+        const [sourceTokenDetails, destinationTokenDetails] = await Promise.all(
+          [
+            getAssetData(sourceToken, currentChainId),
+            getAssetData(destinationToken, currentChainId),
+          ],
+        );
+
+        const result = await smartRouter.getBestQuote(
+          currentChainId,
+          sourceTokenDetails.address,
+          destinationTokenDetails.address,
+          weiAmount,
+        );
+
+        setRoute(result.route);
+        const quote = removeTrailingZerosFromString(
+          fromWei(result.quote.toString()),
+        );
+        setQuote(quote);
+      } catch {
+        setHasQuoteError(true);
+      }
```

Comment:
try/catch can be removed here and error state set to true of entry (`weiAmount`) > 0 and `quote` is zero